airbyte-integrations/connectors/source-mongodb-internal-poc/src/test-integration/java/io/airbyte/integrations/source/mongodb/internal/MongoDbSourceAcceptanceTest.java

```diff
@@ -177,4 +378,38 @@ private void verifyFieldNotExist(final List<AirbyteRecordMessage> records, final
         .isEmpty(), "Records contain unselected columns [%s:%s]".formatted(stream, field));
   }
 
+  private List<AirbyteStateMessage> filterStateMessages(final List<AirbyteMessage> messages) {
+    return messages.stream().filter(r -> r.getType() == AirbyteMessage.Type.STATE).map(AirbyteMessage::getState)
+        .collect(Collectors.toList());
+  }
+
+  private void insertData(final String databaseName, final String collectionName, final int numberOfDocuments) {
+    mongoClient.getDatabase(databaseName).getCollection(collectionName).drop();
+    mongoClient.getDatabase(databaseName).createCollection(collectionName);
+    final MongoCollection<Document> collection = mongoClient.getDatabase(databaseName).getCollection(collectionName);
+    collection
+        .insertMany(IntStream.range(0, numberOfDocuments).boxed().map(this::createDocument).toList());
+  }
+
+  private Document createDocument(final Integer i) {
+    return new Document(NAME_FIELD, "value" + i).append(INT_TEST_FIELD, i);
+  }
+
+  private void validateStateMessages(final List<AirbyteStateMessage> stateMessages) {
+    stateMessages.forEach(stateMessage -> {
+      assertEquals(AirbyteStateMessage.AirbyteStateType.GLOBAL, stateMessage.getType());
+      final AirbyteGlobalState global = stateMessage.getGlobal();
+      assertNotNull(global.getSharedState());
+      assertFalse(global.getSharedState().isEmpty());
```

Comment:
can we validate that the shared state should only contain the specific streams i.e. the first state should not contain the second stream and only contain the first stream, etc.